lib/fs.js

```diff
@@ -2008,31 +2008,53 @@ function ReadStream(path, options) {
   this.bytesRead = 0;
   this.closed = false;
 
-  if (this.start !== undefined) {
-    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
-    }
-    if (this.end === undefined) {
-      this.end = Infinity;
-    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
-    }
+  // Case 0: If end is undefined, set it to default (Infinity)
+  this.end = this.end === undefined ? Infinity : this.end;
 
-    if (this.start > this.end) {
-      const errVal = `{start: ${this.start}, end: ${this.end}}`;
-      throw new ERR_OUT_OF_RANGE('start', '<= "end"', errVal);
-    }
+  // Case 1: If the type of either start or end is not number, throw
+  // ERR_INVALID_ARG_TYPE (TypeError).
+  if (typeof this.start !== 'number') {
```

Comment:
I think you want to allow `undefined` explicitly here and for the other checks as well?

(The `if (this.start !== undefined) {` block from the original variant of this code seems like a pretty reasonable idea, tbh…)

Reply:
Okay. So, should I enclose *all* the type-checking code within that statement?

Reply:
@ryzokuken All that’s related to `start`, yes. `end` is another story, and my original `TODO` was more or less about keeping the `end` checks *out* of such a block :)

Reply:
Exactly. That's why I wanted to keep the checks separate, so they worked either way.

I've enclosed all this code into the `if` block, which *should* work if `end` does not need to be transformed or checked at all if `start` is `undefined`. That probably isn't the case, is it?